### ISSUE TITLE
change-able window size

### ISSUE THREAD

**Issue by RustoMCSpit** [DamRsn/NeuralNote#75]
dont have neuralnote be fixed to one size. weirdly enough updating the window size is the only way to refresh what the gui is displaying on linux with wine

**Comment by RustoMCSpit** [comment]
the "RECORD" button doesn't seem to do anything and the dropdowns don't show the updated selection until a full repaint of the settings window (by closing and opening it again). having it not be able to change size genuinely breaks the program on linux

**Comment by DamRsn** [comment]
I don't plan on making the UI resizable. But if someone knows a simple way to do it and wants to contribute, I'm open.

Also, please stop referring to issues your having with Wine on Linux as it is not and will not be supported.

**Comment by RustoMCSpit** [comment]
> I don't plan on making the UI resizable. But if someone knows a simple way to do it and wants to contribute, I'm open.
> 
> Also, please stop referring to issues you're having with Wine on Linux as it is not and will not be supported.

looking at the source code now, i think the neuralnotemainview is being repainted from memory over and over and thats what causing the bug. im too drunk and tired to actually know if thats whats happening and my understanding of c is shite so apologies if this is a witchhunt but there 100% is something wrong somewhere, i can screen record the bug if needed

**Comment by RustoMCSpit** [comment]
void NeuralNoteMainView::paint(Graphics& g)
{ 
// is this re-painting the display over and over again?
    auto background_image = juce::ImageCache::getFromMemory(BinaryData::background_png, BinaryData::background_pngSize);

**Comment by RustoMCSpit** [comment]
the background image is loaded from memory in the paint method using the juce::ImageCache::getFromMemory function. this means that every time the paint method is called, the image is retrieved from memory.

**Comment by RustoMCSpit** [comment]
im pretty sure but i dont understand this all well enough

**Comment by RustoMCSpit** [comment]
would something like this make more sense

 
 // Initialize the background image to a default image
currentBackgroundImage = juce::ImageCache::getFromMemory(BinaryData::background_png, BinaryData::background_pngSize);

// Use the currentBackgroundImage as the background
// Reusing the loaded image, which was already loaded during object construction.
g.drawImage(currentBackgroundImage, getLocalBounds().toFloat());

**Comment by RustoMCSpit** [comment]
if this is giibberish i apologise i genuinely am off my rocker with whiskey apologies

**Comment by RustoMCSpit** [comment]
> would something like this make more sense

by the way this is designed to allow for future modification of the image 

// this is just pseudo-code and probably doesnt work but you get what im trying to say here
juce::FileChooser fileChooser("Select a background image...",
                               juce::File::getSpecialLocation(juce::File::userPicturesDirectory));

if (fileChooser.browseForFileToOpen()) {
    juce::File selectedImageFile = fileChooser.getResult();
    currentBackgroundImage = juce::ImageFileFormat::loadFrom(selectedImageFile);
}

// redraw new image
repaint();

**Comment by RustoMCSpit** [comment]
if what i said makes no sense apologies, i can screen record the issue if you want to see for yourself

**Comment by RustoMCSpit** [comment]
tl;dr: i think but am not sure that the image is being re-loaded from memory in the paint method using the juce::ImageCache::getFromMemory function. this means that every time the paint method is called, the image is retrieved from memory causing weird graphical issues. i am trying to update the code to load the image once and reuse it in the paint method
